File: src/test/java/cspfj/ParameterManagerTest.java
package cspfj;

import static org.junit.Assert.assertEquals;

import java.math.BigInteger;

import org.junit.Test;

import cspfj.util.Parameter;

public final class ParameterManagerTest {

//	@Parameter("enumTest")
//	private static Enumeration enumTest = Enumeration.A;

	@Parameter("classTest")
	private static Class<?> classTest = Integer.class;
	
	@Parameter("intTest")
	private static int intTest = 12;

	static {
		ParameterManager.register(ParameterManagerTest.class);
	}

//	static enum Enumeration {
//		A, B, C;
//	}
//
//	@Test
//	public void enumTest() {
//		assertEquals(Enumeration.A, enumTest);
//		ParameterManager.parameter("enumTest", Enumeration.B);
//		assertEquals(Enumeration.B, enumTest);
//		ParameterManager.parameterParse("enumTest", "C");
//		assertEquals(Enumeration.C, enumTest);
//	}

//	@Test
//	public void pendingEnumTest() {
//		ParameterManager.parameterParse("pendingEnumTest", "C");
//		assertEquals(Enumeration.C, PendingParameter.pendingEnumTest);
//	}

	@Test
	public void classTest() {
		assertEquals(Integer.class, classTest);
		ParameterManager.parameter("classTest", Double.class);
		assertEquals(Double.class, classTest);
		ParameterManager.parameterParse("classTest", "java.math.BigInteger");
		assertEquals(BigInteger.class, classTest);
	}
	
	@Test
	public void intTest() {
	    assertEquals(12, intTest);
	    ParameterManager.parameter("intTest", 32);
	    assertEquals(32, intTest);
	}

}

File: src/test/scala/cspfj/ParameterManagerTest.scala
package cspfj;

import java.math.BigInteger

import org.junit.Assert.assertEquals
import org.junit.Test

object ParameterManagerTest {
  @Parameter("classTest")
  val classTest = classOf[Int]

  @Parameter("intTest")
  val intTest = 12

  ParameterManager.register(this)
}

final class ParameterManagerTest {

  @Test
  def classTest() {
    assertEquals(classOf[Int], ParameterManagerTest.classTest);
    ParameterManager("classTest") = classOf[Double]
    assertEquals(classOf[Double], ParameterManagerTest.classTest);
    ParameterManager.parse("classTest", "java.math.BigInteger");
    assertEquals(classOf[BigInteger], ParameterManagerTest.classTest);
  }

  @Test
  def intTest() {
    assertEquals(12, ParameterManagerTest.intTest);
    ParameterManager("intTest") = 32
    assertEquals(32, ParameterManagerTest.intTest);
  }

}
